test/integration/import-attributes/pages/ts.ts

```diff
@@ -0,0 +1,5 @@
+import data from '../data' with { type: 'json' }
```

Comment:
eslint is failing
```
/root/actions-runner/_work/next.js/next.js/test/integration/import-attributes/pages/ts.ts
  1:27  error  Parsing error: ';' expected

✖ 1 problem (1 error, 0 warnings)
```

Reply:
I created a separate PR to upgrade typescript-eslint: https://github.com/vercel/next.js/pull/59514

Reply:
@huozhi Should I bump TS to 5.3.3 in this PR too? I'm not sure but it may fix it.

Reply:
Let's bump TS in a separate PR

Reply:
I can do rebase after https://github.com/vercel/next.js/pull/59494 get merged.

Reply:
No, it still can't parse the syntax correctly. I believe we need TypeScript 5.3 to solve the issue.

```
/root/actions-runner/_work/next.js/next.js/test/integration/import-attributes/pages/ts.ts
  1:27  error  Parsing error: ';' expected

✖ 1 problem (1 error, 0 warnings)
```

Reply:
@chentsulin We've upgraded typescript to 5.3 on canary, you can update the branch now 🙏

Reply:
The branch has been updated. 🙏

Reply:
Looks like there are still merge conflicts in pnpm-lock.yaml

Maybe you can try regenerating it from the main branch